src/utils.jl

```diff
@@ -1,3 +1,20 @@
+"""
+    pick_chunksize(::Val{totalsize}, mode, ftype, argtypes...)
+
+Return a reasonable chunk size for batched differentiation.
+
+!!! warning
+    This function is experimental, and not part of the public API.
+"""
+function pick_chunksize(
+    ::Val{totalsize},
```

Comment:
Don't overuse Val like this. If the compiler constant propagates it, it will happen in anyways. If it doesn't this causes dispatch for every possible totalsize parameter.